src/Npgsql/NpgsqlBatchCommand.cs

```diff
@@ -38,6 +38,33 @@ public override string CommandText
     /// <inheritdoc cref="DbBatchCommand.Parameters"/>
     public new NpgsqlParameterCollection Parameters { get; } = new();
 
+#pragma warning disable CA1822 // Mark members as static
+
+#if NET8_0_OR_GREATER
+    /// <inheritdoc/>
+    public override
+#else
+    /// <summary>
+    /// Creates a new instance of an <see cref="System.Data.Common.DbParameter"/> object.
+    /// </summary>
+    /// <returns>A <see cref="System.Data.Common.DbParameter"/> object.</returns>
+    public
+#endif
+    DbParameter CreateParameter() => new NpgsqlParameter();
```

Comment:
Also, as a tiny nit, I'd repeat the signature for the #if - that's what we do for most other APIs I think:

```c#
#if NET8_0_OR_GREATER
    /// <inheritdoc/>
    public override NpgsqlParameter CreateParameter()
#else
    /// <summary>
    /// Creates a new instance of an <see cref="System.Data.Common.DbParameter"/> object.
    /// </summary>
    /// <returns>A <see cref="System.Data.Common.DbParameter"/> object.</returns>
    public NpgsqlParameter CreateParameter()
#endif
    => new NpgsqlParameter();
```

Reply:
Agreed.